### PR TITLE
dbt training instructions + external data source script

### PR DESCRIPTION
**Pull request by chaerinlee1**
This PR contains an updated version of training.md and also SQL scripts that creates an external data source and source tables using Azure SQL and Synapse.

### CODE REVIEW

File: external_azuresql.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   FORMATFILE = 'jaffle-shop/raw_payments.xml',
+   FORMATFILE_DATA_SOURCE = 'JaffleShopBlob'
+) as temp
+go
+-- DROP EXTERNAL DATA SOURCE JaffleShopBlob
+-- go
+
+-- drop table raw.customers
+-- drop table raw.orders
+-- drop table raw.payments
```
Comment (dataders) [inline external_azuresql.sql]: if users complain we can add IF TABLE DOES NOT EXIST... CREATE TABLE. for now we can get rid of this
```suggestion
```

File: training.md [hunk 2]
```diff
@@ -14,14 +14,14 @@ This is the FIRST official training, so apologies for where the training could b
...
 
-
 Ahead of the call if you could please download and install the following (if you haven’t already):
 
 - [Anaconda](https://www.anaconda.com/products/individual)
 - [Microsoft ODBC Driver 17](https://www.microsoft.com/en-us/download/details.aspx?id=56567)
 - [VSCode](https://code.visualstudio.com/Download)
 - [Azure Data Studio](https://docs.microsoft.com/en-us/sql/azure-data-studio/download-azure-data-studio?view=sql-server-ver15)
-- [Azure CLI](https://git-scm.com/downloads)
+- [Azure CLI](https://git-scm.com/downloads) **wrong link, need to fix later**
```
Comment (dataders) [inline training.md]:
```suggestion
- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
```

File: training.md [hunk 3]
```diff
@@ -36,45 +36,123 @@ There may be an issue where your terminal can't find Git or Anaconda, in which c
...
+### Tutorial
+
+#### 0. Connect to External Data Source
+Before we jump in and start creating dbt models, we need to first connect to our external data source and pull our source tables from there. We will be pulling data from a blob storage called `JaffleShopBlob`.
+
+1. Open Azure Data Studio and create a "New Query" by right clicking on the server name and choosing the option.
+2. Paste the Azure SQL code below into the new query and click "Run" at the top. To view the Synapse version of the code click [here](external_synapse.sql).
+   * *should we also have them edit the name of the table? bc everyone will have raw.customers already and there will be an error message when querying*
+
+```
```
Comment (dataders) [inline training.md]:
```suggestion
```sql
```

File: training.md [hunk 4]
```diff
@@ -36,45 +36,123 @@ There may be an issue where your terminal can't find Git or Anaconda, in which c
...
+   4. Edit the file in VSCode and change the `schema` field on line 5 to your initials or anything that serves as an identifier for your work.
+4. Create your conda environment and install the necessary packages in here **once** so that you can reuse this environment for future dbt work.
+   1. Open a new terminal by clicking on "Terminal" at the top and "New Terminal"
+   2. Click "Allow" when a pop up appears asking if you allow the workspace to modify your terminal shell.
+      * If you see "PS" next to your current directory on the command line, that means you're using powershell and we don't want to use that. The fix: open a new terminal again and you should now be using cmd.
+   3. Since you already installed Anaconda, you should see `conda activate base` run automatically and (base) should be next to your current directory.
+   4. Create a new, empty environment by running `conda create -n dbtenv python=3.7.9`. The name of your environment is `dbtenv` and we are using Python version 3.7.9.
+      * *do we want to include this or should we just have them install in the base environment?*
```
Comment (dataders) [inline training.md]: we discussed leaving as is for now, but maybe observing users go through tutorial later.
```suggestion
```

File: training.md [hunk 5]
```diff
@@ -89,34 +167,44 @@ Let's make some dbt models! The following steps will show how `dim_customers`, a
...
-## 2. Learning dbt from better teachers than Anders
-
-Now you're cut loose to learn from the best. [Here's a video](https://www.loom.com/share/c0e55d1734c849c184b1dd3b7df83f02) of me giving a partial intro to these resources
+<!-- Now you're cut loose to learn from the best. [Here's a video](https://www.loom.com/share/c0e55d1734c849c184b1dd3b7df83f02) of me giving a partial intro to these resources
 
-here's where our special [`dim_customers.sql`](dim_customers.sql) lives (theirs doesn't work on TSQL)
+here's where our special [`dim_customers.sql`](dim_customers.sql) lives (theirs doesn't work on TSQL) -->
 
 - [Kyle's amazing course](https://courses.getdbt.com)
 - [Claire's classic text + video walkthrough](https://docs.getdbt.com/tutorial/setting-up)
```
Comment (dataders) [inline training.md]: is there a way we could more cleanly hand off people into Claire's (or Kyle's) tutorial? also, delete the commented out section probably

**Review by dataders — CHANGES_REQUESTED** [review]
strong work! just a few small changes

File: training.md [hunk 6]
```diff
@@ -36,45 +36,123 @@ There may be an issue where your terminal can't find Git or Anaconda, in which c
...
+
+### Azure Data Studio
+
+1. Last step in the set-up is to log in to the database
+   1. Open Azure Data Studio and click on the first icon on the left side panel called "Connections".
+   2. Click "Add Connection" and paste in the connection info below in the relevant fields. This info can also be found in `profiles.yml`
+      * Server: dbtavatrain.database.windows.net
+      * Authentication type: Azure Active Directory
+      * Account: Click the drop down menu, then "Add an account" and sign in like before
+      * Azure AD tenant: Avanade
```
Comment (chaerinlee1) [inline training.md:82]: Will the people you speak with this week have Avanade credentials? Same question applies to line 69.

File: training.md [hunk 7]
```diff
@@ -36,45 +36,122 @@ There may be an issue where your terminal can't find Git or Anaconda, in which c
...
+Try the walk-through tutorial of using dbt below. There are also additional resources found at the end of this section as well. Happy modeling!
+
+### Tutorial
+
+#### 0. Connect to External Data Source
+Before we jump in and start creating dbt models, we need to first connect to our external data source and pull our source tables from there. We will be pulling data from a blob storage called `JaffleShopBlob`.
+
+1. Open Azure Data Studio and create a "New Query" by right clicking on the server name and choosing the option.
+2. Paste the Azure SQL code below into the new query and click "Run" at the top. To view the Synapse version of the code click [here](external_synapse.sql).
+   * *should we also have them edit the name of the table? bc everyone will have raw.customers already and there will be an error message when querying*
```
Comment (chaerinlee1) [inline training.md]: can you refresh my memory if you want everyone going through the tutorial to be able to run this code, or if you just want them to look at it and understand how we're connecting to the external data source?

File: training.md [hunk 8]
```diff
@@ -129,4 +216,4 @@ here's where our special [`dim_customers.sql`](dim_customers.sql) lives (theirs
 - [dbt-utils](https://github.com/fishtown-analytics/dbt-utils)
 - production workflow
 
-## 4. how can it help Avanade?
\ No newline at end of file
+## 4. how can it help Avanade?
```
Comment (chaerinlee1) [inline training.md]: do we still need this section?

File: training.md [hunk 9]
```diff
@@ -89,38 +166,44 @@ Let's make some dbt models! The following steps will show how `dim_customers`, a
...
-
-here's where our special [`dim_customers.sql`](dim_customers.sql) lives (theirs doesn't work on TSQL)
-
-- [Kyle's amazing course](https://courses.getdbt.com)
-- [Claire's classic text + video walkthrough](https://docs.getdbt.com/tutorial/setting-up)
-- [thorough docs worth going through](https://docs.getdbt.com/docs/building-a-dbt-project/projects)
+- [Claire's classic text + video walkthrough](https://docs.getdbt.com/tutorial/setting-up) - check this out if you want to learn how to set up and deploy a dbt project using BigQuery (flip through the modules under "Getting Started" on the left)
+- [Kyle's amazing course](https://courses.getdbt.com) - this course shows you how to use dbt cloud and goes over some fundamental concepts around dbt
+- [thorough docs worth going through](https://docs.getdbt.com/docs/building-a-dbt-project/projects) - these are great overviews about modeling, testing, documentation, sources, and other additional topics we did not cover (flip through the topics under "Building a dbt Project" on the left)
```
Comment (chaerinlee1) [inline training.md:205]: is this better?

File: training.md (hunk 8 above)
Comment (dataders) [reply training.md]: great question. probably not!
```suggestion
```

File: training.md (hunk 7 above)
Comment (dataders) [reply training.md]: if you're making your own db and not using a db that has been made for them, then they'll need to run this script to create the three raw tables

File: training.md (hunk 9 above)
Comment (dataders) [reply training.md:205]: I guess i was thinking that we could link them directly to one of these pages, to save you time in re-writing them here on this page. but looking at them now, they don't have as much information as i remember. most of the info is in the videos anyway. and do we want them making their own repo??
https://docs.getdbt.com/tutorial/create-a-project-dbt-cli
https://docs.getdbt.com/tutorial/build-your-first-models

File: training.md (hunk 6 above)
Comment (dataders) [reply training.md:82]: nope! they'll have to use their own database and database credentials. In a future PR, we should make a tutorials for how to do that with Azure SQL

**Review by dataders — APPROVED** [review]
excellent work! I'm sure FT would love to see this once after a few revisions.

**Review by dataders — APPROVED** [review]

File: training.md [hunk 10]
```diff
@@ -36,45 +36,121 @@ There may be an issue where your terminal can't find Git or Anaconda, in which c
...
+INTO raw.payments
+FROM OPENROWSET(
+   BULK  'jaffle-shop/raw_payments.csv',
+   DATA_SOURCE = 'JaffleShopBlob',
+   FORMAT ='CSV',
+   FORMATFILE = 'jaffle-shop/raw_payments.xml',
+   FORMATFILE_DATA_SOURCE = 'JaffleShopBlob'
+) as temp
+go
+```
```
Comment (chaerinlee1) [inline training.md:147]: do you think the description at the top is clear?

**Pull request merged**